recipe/build.sh

```diff
@@ -1,27 +1,41 @@
-#! /bin/bash
-
-set -e
-
-cmake_args=(
-    -DCMAKE_BUILD_TYPE=Release
-    -DCMAKE_COLOR_MAKEFILE=OFF
-    -DCMAKE_INSTALL_PREFIX=$PREFIX
-)
-
-if [ $(uname) = Darwin ] ; then
-    cmake_args+=(
-        -DCMAKE_CXX_FLAGS="$CXXFLAGS -stdlib=libc++"
-	-DCMAKE_OSX_DEPLOYMENT_TARGET=$MACOSX_DEPLOYMENT_TARGET
-	-DCMAKE_OSX_SYSROOT=/
-    )
-fi
-
-mkdir build
-cd build
-cmake "${cmake_args[@]}" ..
-make -j${CPU_COUNT} ${VERBOSE_CM}
-# make test -- these do not pass
-make install
+#!/bin/bash
+echo "Building ${PKG_NAME}."
+
+
+# Isolate the build.
+mkdir -p Build-${PKG_NAME}
+cd Build-${PKG_NAME} || exit 1
+
+
+# Generate the build files.
+echo "Generating the build files..."
+cmake .. ${CMAKE_ARGS} \
+      -GNinja \
+      -DCMAKE_PREFIX_PATH=$PREFIX \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_BUILD_TYPE=Release \
+
+
+# Build.
+echo "Building..."
+ninja -j${CPU_COUNT} || exit 1
+
+
+# Perform tests.
+#  echo "Testing..."
+#  ninja test || exit 1
+#  path_to/test || exit 1
+#  ctest -VV --output-on-failure || exit 1
```

Comment:
I think there's value in running the author's regression tests... What happens when we do?

Reply:
Added the tests in a new commit.
The result is commented out because some tests are failing.